data_structures/src/chain.rs

```diff
@@ -183,9 +183,111 @@ impl From<Sha256> for Hash {
 pub type SHA256 = [u8; 32];
 
 /// Transaction data structure
-// FIXME(#99): define Transaction as defined in issue
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct Transaction {
+    pub version: u32,
+    pub inputs: Vec<Input>,
+    pub outputs: Vec<Output>,
+    pub signatures: Vec<KeyedSignature>,
+}
+
+/// Input data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub enum Input {
+    ValueTransfer(ValueTransferInput),
+    Commit(CommitInput),
+    Reveal(RevealInput),
+    Tally(TallyInput),
+}
```

Comment:
This is just a suggestion, but why don't we just use the enum directly instead of wrapping its field in another struct? Like this:

```rust
#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
pub enum Input {
    ValueTransfer { transaction_id: [u8; 32], output_index: u32 },
    Commit { ... },
    Reveal { ... },
    Tally { ... },
}
```
 It will spare us a few lines for defining the struct and deriving.

Reply:
We had the same problem with `Command`. The main advantage is that we can define the types outside and reuse them. :)